src/js/quick-hull-3d/quick-hull-3d.js

```diff
@@ -0,0 +1,447 @@
+/**
+ * Creates a convex hull object.
+ * If set of points is specialized initializes it to the convex hull.
+ */
+var QuickHull3D = function(points) {
+    this.pointBuffer = [];
+
+    this.minVertices = [];
+    this.maxVertices = [];
+
+    this._claimed = new VertexList();
+
+    if (points !== void 0) {
+        this.build(points);
+    }
+};
+
+QuickHull3D.POINT_RELATIVE = 0x8;
+QuickHull3D.INDEXED_FROM_ZERO = 0x4;
+QuickHull3D.INDEXED_FROM_ONE = 0x2;
+
+QuickHull3D.prototype.debug = false;
+QuickHull3D.prototype.getDebug = function() {};
+QuickHull3D.prototype.setDebug = function() {};
+QuickHull3D.prototype.getDistanceTolerance = function() {};
+QuickHull3D.prototype.setExplicitDistanceTolerance = function(tolerance) {};
+QuickHull3D.prototype.getExplicitDistanceTolerance = function() {};
+QuickHull3D.prototype.addPointToFace = function(vertex, face) {};
+QuickHull3D.prototype.removePointFromFace = function(vertex, face) {};
+QuickHull3D.prototype.removeAllPointFromFace = function(face) {};
+
+/**
+ * Constructs the convex hull of a set of points
+ */
+QuickHull3D.prototype.build = function(points, numberOfPoints) {
```

Comment:
4 versions of ```build```: for point and for coordinates, with and without number of points: https://github.com/CindyJS/CindyJS/blob/master/src/java/cindyjs/quickhull3d/QuickHull3D.java#L451-L534